recirq/qcqmc/hamiltonian.py

```diff
@@ -0,0 +1,402 @@
+from pyscf import ao2mo, fci, gto, scf
+
+from recirq.qcqmc import config, data
+
+# xyz coordinates of an atom
+PositionT = Tuple[float, float, float]
+# [atom name, xyz]
+GeomT = Tuple[str, PositionT]
+
+
+def compute_integrals(
+    pyscf_molecule: gto.Mole, pyscf_scf: Union[scf.RHF, scf.ROHF]
+) -> Tuple[np.ndarray, np.ndarray]:
+    """Compute the (restricted) 1-electron and 2-electron integrals in the spatial orbital basis.
+
+    From: openfermionpyscf _run_pyscf.
+
+    Args:
+        pyscf_molecule: A pyscf Mole instance.
+        pyscf_scf: A PySCF "SCF" calculation object.
+
+    Returns:
+        one_electron_integrals: An N by N array storing h_{pq}
+        two_electron_integrals: A 4-D tensor storing h_{pqrs} in using the openfermion convention.
+    """
+    # Get one electrons integrals.
+    n_orbitals = pyscf_scf.mo_coeff.shape[1]
+    one_electron_compressed = functools.reduce(
+        np.dot, (pyscf_scf.mo_coeff.T, pyscf_scf.get_hcore(), pyscf_scf.mo_coeff)
+    )
+    one_electron_integrals = one_electron_compressed.reshape(
+        n_orbitals, n_orbitals
+    ).astype(float)
+
+    # Get two electron integrals in compressed format.
+    two_electron_compressed = ao2mo.kernel(pyscf_molecule, pyscf_scf.mo_coeff)
+
+    two_electron_integrals = ao2mo.restore(
+        1, two_electron_compressed, n_orbitals  # no permutational symmetry
+    )
+    # See PQRS convention in OpenFermion.hamiltonians._molecular_data
+    # h[p,q,r,s] = (ps|qr)
+    two_electron_integrals = np.asarray(
+        two_electron_integrals.transpose(0, 2, 3, 1), order="C"
+    )
+
+    return one_electron_integrals, two_electron_integrals
+
+
+@attrs.frozen
+class HamiltonianFileParams(data.Params):
+    """Class for storing the parameters that specify loading hamiltonian integrals from a file.
+
+    Args:
+        name: A descriptive name for the hamiltonian.
+        integral_key: A string specifying the directory in data/integrals where
+            the integrals should live.
+        n_orb: The number of spatial orbitals.
+        n_elec: The total number of electrons.
+        do_eri_restore: Whether to restore 8-fold symmetry to the integrals.
+        path_prefix: An optional path string to prepend to the default
+            hamiltonian data directory (by default the integrals will be written /
+            read to ./data/hamiltonians.)
+    """
+
+    name: str
+    integral_key: str
+    n_orb: int
+    n_elec: int
+    do_eri_restore: bool = False
+    path_prefix: str = ""
+
+    def _json_dict_(self):
+        return attrs.asdict(self)
+
+    @property
+    def path_string(self) -> str:
+        return (
+            self.path_prefix + config.OUTDIRS.DEFAULT_HAMILTONIAN_DIRECTORY + self.name
+        )
+
+
+@attrs.frozen
+class PyscfHamiltonianParams(data.Params):
+    """Class for describing the parameters to get a Hamiltonian from Pyscf.
+
+    name: A name for the Hamiltonian
```

Comment:
Indentation is a little messed up here.  Should begin with "Args:" and the rest should be indented 4 spaces.

Reply:
Whoops